W-11862525-first-draft-for-beta-site-build-CP (#114)
mulesoft/docs-apikit: modules/ROOT/pages/apikit-4-display-fault-task.adoc

= Display a Typed Fault
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> Make the `ListInventory` operation display the typed fault. In the tshirt2.wsdl file, there is one typed fault named *TshirtFault*, which is mapped to the operations that the WSDL has:

Reviewer: Is there a way to make this more of a short description. It feels like a part of a tutorial. Can you give more context as to why do this?

> Make the `ListInventory` operation display the typed fault. In the tshirt2.wsdl file, there is one typed fault named *TshirtFault*, which is mapped to the operations that the WSDL has:

Reviewer: ```suggestion
Make the `ListInventory` operation display the typed fault. The tshirt2.wsdl file includes the *TshirtFault* typed fault, which is mapped to the WSDL operations:
```

Author: OK! Thanks!

* `OrderTshirt`
* `ListInventory`
* `TrackOrder`

APIkit for SOAP supports the following types of SOAP faults:

* Soap Fault 1.1
* Soap Fault 1.2

You can select either type by using any WSDL file, even for files that do not contain typed faults.

. In Anypoint Studio, search for *fault* in the command palette.
+

> Multiple `SOAP Fault` components can appear.

Reviewer: Is this even necessary?

Author: This is relevant because some `SOAP Fault`components may trigger an error and the user should select another SOAP Fault component.
Please, let me know if you think we can do without this sentence.

+

> . Drag and drop the `SOAP Fault` component into the `ListInventory:api-config` flow.

Reviewer: ```suggestion
. Drag `SOAP Fault` to the `ListInventory:api-config` flow.
```

Reviewer: https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-%22draganddrop%22

Author: OK! Thanks!

+
If *Operation Unavailable* message appears, cancel the operation and select another `SOAP Fault` component.

> . Repeat these steps until you find a version match.

Reviewer: This seems like a terrible UX! :)

Author: I agree! Requesting the customer to select as many SOAP Fault components are available until one works does not seem to be the great journey we are looking for the user.
